### PR TITLE
feat(evm): allow SetWormholeAddress without dual-verify

### PR DESCRIPTION
**Pull request by keyvankhademi**
## Summary
- Remove the post-write dual-VAA re-verify from `SetWormholeAddress` so a governance VAA verified on the legacy Wormhole can point `_state.wormhole` at a Pro receiver whose guardians cannot verify that same VAA.
- Reject zero-address and empty-code targets (`extcodesize`) instead; emit `WormholeAddressSet` as before.
- Add forge tests for an unrelated receiver, reject zero/no-code, and the `SetDataSources` → `SetWormholeAddress` (and upgrade-first) cutover sequence. Governance emitter and sequence watermark stay unchanged.

This is PR A of the EVM pro-compatible in-place cutover. `PythUpgradable` runtime is **24,067 bytes** (EIP-170 limit 24,576).

## Test plan
- [ ] `cd target_chains/ethereum/contracts && forge test --match-contract PythGovernanceTest`
- [ ] `forge build --sizes` — confirm `PythUpgradable` runtime stays under 24,576
- [ ] Confirm `testSetWormholeAddress` (same-guardian) still passes
- [ ] Confirm `testSetWormholeAddressToUnrelatedReceiver` succeeds (different guardian keys; VAA signed by the original set)
- [ ] Confirm zero address and `address(0xdead)` revert with `InvalidWormholeAddressToSet`

---

<a href="https://app.devin.ai/review/pyth-network/pyth-crosschain/pull/3973" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

File: target_chains/ethereum/contracts/test/PythGovernance.t.sol [hunk 1]
```diff
@@ -264,6 +264,228 @@ contract PythGovernanceTest is
...
+        return
+            abi.encodePacked(
+                MAGIC,
+                uint8(GovernanceModule.Target),
+                uint8(GovernanceAction.SetDataSources),
+                TARGET_CHAIN_ID,
+                uint8(1), // Number of data sources
+                uint16(1), // Chain ID
+                bytes32(
+                    0x0000000000000000000000000000000000000000000000000000000000001111
```
Comment (guibescos) [inline target_chains/ethereum/contracts/test/PythGovernance.t.sol]: extract this into a constant?

File: target_chains/ethereum/contracts/test/PythGovernance.t.sol [hunk 2]
```diff
@@ -264,6 +264,228 @@ contract PythGovernanceTest is
...
+            TEST_GOVERNANCE_CHAIN_ID,
+            TEST_GOVERNANCE_EMITTER,
+            3
+        );
+        PythGovernance(address(pyth)).executeGovernanceInstruction(
+            setWormholeVaa
+        );
+
+        assertCutoverState(newWormhole, 3);
+    }
```
Comment (guibescos) [inline target_chains/ethereum/contracts/test/PythGovernance.t.sol:528]: I'd feel a little bit better about this test if we executed something with the new Wormhole

**Review by guibescos — APPROVED** [review]

File: target_chains/ethereum/contracts/test/PythGovernance.t.sol (hunk 2 above)
Comment (keyvankhademi) [reply target_chains/ethereum/contracts/test/PythGovernance.t.sol:528]: done

File: target_chains/ethereum/contracts/test/PythGovernance.t.sol (hunk 1 above)
Comment (keyvankhademi) [reply target_chains/ethereum/contracts/test/PythGovernance.t.sol]: done
